nixos/modules/services/networking/cgproxy.nix

```diff
@@ -0,0 +1,110 @@
+
+      settings = mkOption {
+        description = lib.mdDoc ''
+          options priority:
+          program_noproxy > program_proxy > cgroup_noproxy > cgroup_proxy
+          enable_ipv6 = enable_ipv4 > enable_dns > enable_tcp = enable_udp
+          command cgproxy and cgnoproxy always have highest priority
+
+          Note: cgroup in configuration need to be exist, otherwise ignored
+        '';
+        default = { };
+        type = types.submodule {
+          freeformType = settingsFormat.type;
+          options = {
+            port = mkOption {
+              type = types.port;
+              default = 12345;
+              defaultText = literalExpression "12345";
+              description = lib.mdDoc "tproxy listenning port";
+            };
+            program_noproxy = mkOption {
+              type = types.listOf types.str;
+              default = [ "v2ray" "qv2ray" ];
+              defaultText = literalExpression ''[ "v2ray" "qv2ray" ]'';
+              description = lib.mdDoc ''
+                program that won't be proxied
+              '';
+            };
+            program_proxy = mkOption {
+              type = types.listOf types.str;
+              default = [ ];
+              defaultText = literalExpression "";
+              description = lib.mdDoc ''
+                program need to be proxied
+              '';
+            };
+            cgroup_noproxy = mkOption {
+              type = types.listOf types.str;
+              default = [ "/system.slice/v2ray.service" ];
+              defaultText = literalExpression ''"/system.slice/v2ray.service"'';
+              description = lib.mdDoc ''
+                cgroup array that no need to proxy, `/noproxy.slice` is preserved
+              '';
+            };
+
+            cgroup_proxy = mkOption {
+              type = types.listOf types.str;
+              default = [ ];
+              defaultText = literalExpression "";
+              description = lib.mdDoc ''
+                cgroup array that need to proxy, `/proxy.slice` is preserved
+              '';
+            };
+
+            enable_gateway = mkOption {
+              type = types.bool;
+              default = false;
+              defaultText = literalExpression "false";
+              description = lib.mdDoc ''
+                enable gateway proxy for local devices
+              '';
+            };
+
+            enable_dns = mkOption {
+              type = types.bool;
+              default = true;
+              defaultText = literalExpression "true";
+              description = lib.mdDoc ''
+                enable dns to go to proxy
+              '';
+            };
+          };
+
+        };
+      };
+
+      package = mkPackageOptionMD pkgs "cgproxy" { };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ cfg.package ];
+
+    environment.etc."cgproxy/config.json".source = settingsFile;
+
+    systemd.packages = [ cfg.package ];
```

Comment:
Why do we need this?

Reply:
> Why do we need this?

Here in the package contains a service unit file from upstream, that provides autostart while the network is online.

Reply:
If you just want the service file, then systemd.packages should be used.